### ISSUE TITLE
[IMPORTANT BUG] This mod spams the text log every millisecond, resulting in an extremely huge file (over 20GBs)

### ISSUE THREAD

**Issue by Sul4urx** [mega12345mega/NBT-Editor#89]
This mod spams the following text about 7500 times per second in `.minecraft/logs/latest.log`: (There is more that is spams as well)
```
[<time>] [NBTEditor/Async/HeadRefresh/Auto/ERROR]: Failed to fetch heads (no-cache) | Stack Trace:
[<time>] [NBTEditor/Async/HeadRefresh/Auto/ERROR]: java.net.UnknownHostException: minecraft-heads.com
[<time>] [NBTEditor/Async/HeadRefresh/Auto/INFO]: Attempting fallback provider for: alphabet
[<time>] [NBTEditor/Async/HeadRefresh/Auto/ERROR]: Failed to fetch heads for alphabet! (OF)
[<time>] [NBTEditor/Async/HeadRefresh/Auto/ERROR]: java.net.UnknownHostException: heads.pages.dev
[<time>] [NBTEditor/Async/HeadRefresh/Auto/ERROR]: Failed to update database! Check above for any errors.
```
When it does that, It creates an extremely huge file (it should make the log file over 20 GBs for each hour of playing minecraft).

**Comment by mega12345mega** [comment]
Obviously a timeout should be added when re-trying, but since this also relates to an issue that used to happen more often, did you happen to be offline when running the mod? The main head provider, minecraft-heads.com, seemed to frequently go offline. That's why there is now a fallback (heads.pages.dev), so I haven't seen this error in a while. I wonder if both went down at the same time, or if you were simply not connected to the internet.

**Comment by Sul4urx** [comment]
> Obviously a timeout should be added when re-trying, but since this also relates to an issue that used to happen more often, did you happen to be offline when running the mod? The main head provider, minecraft-heads.com, seemed to frequently go offline. That's why there is now a fallback (heads.pages.dev), so I haven't seen this error in a while. I wonder if both went down at the same time, or if you were simply not connected to the internet.

I have of course turned on airplane mode, so that means I didn't have an internet connection. But either way this should not be spammed and only be warned upon loading the game, not every millisecond.

**Comment by mega12345mega** [comment]
A dev build with a fix for this has been available for a while now, I just had forgotten to label the issue

**Comment by Sul4urx** [comment]
> A dev build with a fix for this has been available for a while now, I just had forgotten to label the issue

A dev build? Wait isn't fixing this problem just like changing a few lines of code?

**Comment by mega12345mega** [comment]
It is - basically, each update consists of many changes, and I "publish" each change on Github before releasing all of them at once officially. Dev builds don't typically work on versions of Minecraft before the last supported version (so right now, dev builds only support 1.20.4). Getting it working on 1.17 involves more changes also published to Github. All of this takes place on the dev branch, before getting merged into main once I decide the update is ready. (Think of "dev builds" like Minecraft snapshots.)

**Comment by mega12345mega** [comment]
I've now fixed this in v2.0.0!
